src/hooks/actions.tsx

```diff
@@ -292,6 +217,67 @@ export function useActions(
     ),
   ];
 
+  useEffect(() => {
```

Comment:
this is pretty much exactly how the voter dapp does it, only difference is now all the handlers live here instead of in individual contract call handler functions